Smarter autosigner

diranged opened the pull request:
Instead of being dead-set on the exact layout of the CSR, search the CSR
properly for the bits and then compare them. No need to ensure that the
CSR has exactly the right number of attributes... nor to assume the exact
order in which they will show up.

nehalrp commented on lib/etc/autosign.rb:
```diff
@@ -78,19 +79,14 @@ def initialize()
-      end
+      @log.info("Searching for #{key}")
+      attr = attributes.find{|x| x.oid == key}
+      attr.value.first.value
```
Unless you're at the very end of the method, you should explicitly say `return attr.value.first.value`.

nehalrp commented on lib/etc/autosign.rb:
```diff
@@ -78,19 +79,14 @@ def initialize()
   #   - The value of the attribute matching the pp_preshared_key name.
   #
-  def find_preshared_key(attributes)
+  def find_attribute(key, attributes)
```
You'll need to update the rspecs:
https://github.com/Nextdoor/puppet_rightscale/blob/matt/better_autosigner/spec/unit/etc/autosigner_spec.rb

nehalrp commented on lib/etc/autosign.rb:
```diff
@@ -78,19 +79,14 @@ def initialize()
+      attr.value.first.value
     rescue Exception => e
-      raise("Invalid CSR: #{e}")
+      # Pass
```
We can get rid of exception handling here entirely if we're looking for keys? We'd have to handle `find_attribute ` returning `nil` then.

diranged commented:
Ok I've fixed up all the broken tests - and refactored a bit more. Going to merge this as is, knowing that I hope to kill the autosigner entirely soon, or at least refactor it completely into a bash-script that uses `rsc`.
